### DISCUSSION TITLE
Water bodies

### DISCUSSION THREAD

**Discussion by AntoniaJost** [CliDyn/climsight#31] [General]
I've encountered specific challenges related to accurately identifying and handling clicks on water bodies, in particular oceans, lakes, and rivers. This discussion gives an overview of the potential implications and areas for future improvements.

1. **Handling ocean selection**: When an oceanic point (= not land) is selected, we currently do not execute the get_location and get_address functions, as this would cause the Nominatim API to crash. This approach effectively prevents system errors but also limits our analyses to land areas as we cannot get positions in open water. 
2. **Address fetching for lakes and rivers**: Currently, Nominatim cannot classify water bodies, such as lakes and rivers. This results in Nominatim providing the nearest available address or postal code for clicks on these water bodies, which can sometimes lead to unexpected address details, like specific house numbers for river locations. Although not perfect for detailed location accuracy, this approach still allows for broader analyses that are based on a location, such as natural hazard assessments, without major issues. Our identification of most rivers and lakes relies on Natural Earth data, which still effectively informs users about the nature of their selected point.
4. **Incompleteness of river and lake data sets**: We've observed that the river and lake data sets are not exhaustive, with many smaller water bodies not being recognized due to their absence in the source file. While these obmissions do not cause errors in the fetching process, they represent a gap in out geospatial analysis capabilities. 
5. **Discrepancy between data sets**: A notable challenge arises from discrepancies in coastal definitions between the Natural Earth data and OpenStreetMap, despite both using the WGS84 projection system. These can lead to inaccuracies, such as incorrectly identifying land points as oceanic. An example of this issue is when a point in the Weser, close to Bremerhaven, is inaccurately classified as "being in the ocean", thus preventing address retrieval. Tackling this issue is complex, as we rely on different data sets for water/land information which inevitably introduces potential for variance in coastline representation.

**Comment by kuivi** [comment]
2 and 3:
We can use the Global Self-consistent, Hierarchical, High-resolution Geography Database (GSHHG) to determine if a point is located within any water body. GSHHG includes numerous shapefiles, such as the boundary between land and ocean or lake and land, in addition to ten classes of rivers. We can utilize information such as the area of the water body and so on, but it does not include names. All this information is available at six different resolutions, enabling offline use without any API.
4: Is is rather strange. OpenStreetMap from my experience is rather accurate. We can try use GSHHG to identify  land/water mask.
unfortunately there are also some discrepancy between OpenStreetMap / GSHHG and satellite images. but usually they are  minor.

**Comment by AntoniaJost** [comment]
Regarding 4: This should be less of an issue once PR #32 is merged, because here I switched to using the 10m coastline instead of 50m, which was causing most of the discrepancy. It might still not be perfect, but much better at least.

**Comment by koldunovn** [comment]
The GSHHG is definitely something to explore. I merged the https://github.com/koldunovn/climsight/pull/32 - I suggest we live for a while with this solution and then explore if it can be improved.

**Comment by kuivi** [comment]
Currently, loading and combining all shapefiles takes a lot of time, approximately 10 seconds. I suggest that the shapefiles should be preloaded and combined in the initial step.
